docs/tasks/debug-application-cluster/logging-stackdriver.md

```diff
@@ -5,24 +5,104 @@ assignees:
 
-After creation, you can discover logging agent pods in the `kube-system` namespace,
-one per node, by running the following command:
+### GKE
+
+Stackdriver is the default logging solution for clusters deployed on Google Container Engine.
+Stackdriver Logging is deployed by default unless you explicitly opt-out.
+
+### Deploying with a new cluster
+
+To deploy Stackdriver Logging on a *new* cluster that you're
+creating using `kube-up.sh`, do the following:
+
+1. Set the `KUBE_LOGGING_DESTINATION` environment variable to `gcp`.
+1. **If not running on GCE**, include the `beta.kubernetes.io/fluentd-ds-ready=true`
+in the `KUBE_NODE_LABELS` variable.
+
+Once your cluster has started, each node should be running the Stackdriver Logging agent.
+The `DaemonSet` and `ConfigMap` are configured as addons.
+
+### Deploying to an existing cluster
+
+1. Apply a label on each node, if not already present.
+
+    The Stackdriver Logging agent deployment uses node labels to determine to which nodes
+    it should be allocated. You can ensure that your nodes are labelled properly
+    by running `kubectl describe` as follows:
+
+    ```shell
+    kubectl describe node $NODE_NAME
+    ```
+
+    The output should be similar to this:
+
+    ```
+    Name:           NODE_NAME
+    Role:
+    Labels:         beta.kubernetes.io/fluentd-ds-ready=true
+    ...
+    ```
+
+    Ensure that the output contains the label `beta.kubernetes.io/fluentd-ds-ready=true`. If it
+    is not present, you can add it using the `kubectl label` command as follows, for each node:
+
+    ```shell
+    kubectl label node $NODE_NAME beta.kubernetes.io/fluentd-ds-ready=true
+    ```
+
+    **Note:** If a node fails and has to be recreated, you must re-apply the label to
+    the recreated node. To make this easier, you can use Kubelet's command-line parameter
+    for applying node labels in your node startup script.
+
+1. Deploy a `ConfigMap` with the logging agent configuration by running the following command:
+
+    ```shell
+    kubectl create -f https://raw.githubusercontent.com/kubernetes/kubernetes/release-1.6/cluster/addons/fluentd-gcp/fluentd-gcp-configmap.yaml
+    ```
+
+    The command creates the `ConfigMap` in the `kube-system` namespace. You can download the file
+    manually and change it before creating the `ConfigMap` object.
+
+1. Deploy the logging agent `DaemonSet` by running the following command:
+
+    ```shell
+    kubectl create -f https://raw.githubusercontent.com/kubernetes/kubernetes/release-1.6/cluster/addons/fluentd-gcp/fluentd-gcp-ds.yaml
+    ```
+
+    You can download and edit this file before using it as well.
+
+## Verifying your Logging Agent Deployment
+
+After Stackdriver DaemonSet is deployed, you can discover logging agent pods
+in the `kube-system` namespace, one per node, by running the following command:
 
 ```shell
-$ kubectl get pods --namespace=kube-system
+kubectl get pods --namespace=kube-system
```

Comment:
Pods running on the node is the e2e signal that everything's working. DS might be present, but without any pods scheduled.